site/src/utils/appearance.ts

```diff
@@ -0,0 +1,16 @@
+export const getApplicationName = (): string => {
```

Comment:
This is something I might not be fully keyed into, but are the meta tags able to change over time?

I guess the best to frame my concerns is that these functions make perfect sense in vanilla JS, but in React specifically, `querySelector` is considered an escape hatch that shouldn't be called directly within render logic. If the meta tags cannot ever change over time, then I think this is a rule we can afford to break a little, but if they can, I think a better approach might be to turn these into hooks, to make sure React properly re-renders after any changes happen

Reply:
I will try to give you as much context as possible :)

> This is something I might not be fully keyed into, but are the meta tags able to change over time?

Yes, as we are rendering them here: [code](https://github.com/coder/coder/blob/main/site/site.go#L293)

> querySelector is considered an escape hatch that shouldn't be called directly within render logic.

I followed the pattern I found in the codebase, for instance [buildInfo](https://github.com/coder/coder/blob/main/site/src/api/queries/buildInfo.ts#L5)

> If the meta tags cannot ever change over time, then I think this is a rule we can afford to break a little, but if they can, I think a better approach might be to turn these into hooks, to make sure React properly re-renders after any changes happen

AFAIK the user has to reload the page to see changed meta tags, there is nothing like a dynamic change or channel subscription.

Let me know your thoughts!

Reply:
Sorry, I should've been more clear. When I meant "change over time", I meant "is the single-page client app able to update the meta tag dynamically, or would it basically be locked in after the first request – at least until a full page refresh?"

If the answer's no, I think we can get away with not having the hooks logic. But if it can be, I feel like we would want to have a hook for this, probably using the vanilla DOM's `MutationObserver` API

Reply:
> would it basically be locked in after the first request – at least until a full page refresh?

yes! This is how I expect it to be.

Reply:
Okay, so in that case, I think that this is a case where we're not following React's guidelines 100%, but this shouldn't ever break on us

The only thing that might be worth changing is updating the `document.querySelector` to `document.head.querySelector`, just to minimize the query-able area as much as possible, since we know we're only going to be accessing `<meta>` elements. But that might be an over-optimization